crates/forge/tests/e2e/docs_snippets_validation.rs

```diff
@@ -1,43 +1,62 @@
 use clap::Parser;
-use docs::validation::{
-    assert_valid_snippet, extract_snippets_from_directory, get_parent_dir,
-    print_skipped_snippet_message, print_success_message,
+use docs::snippet::SnippetType;
+use docs::utils::{
+    assert_valid_snippet, get_parent_dir, print_skipped_snippet_message, print_success_message,
 };
+use docs::validation::extract_snippets_from_directory;
 use forge::Cli;
-use regex::Regex;
+use shared::test_utils::output_assert::assert_stdout_contains;
+
+use super::common::runner::{setup_package, test_runner};
+
 #[test]
 fn test_docs_snippets() {
     let root_dir = get_parent_dir(2);
     let docs_dir = root_dir.join("docs/src");
 
-    let re = Regex::new(r"(?ms)```shell\n\$ (snforge .+?)\n```").expect("Invalid regex pattern");
-
-    let snippets = extract_snippets_from_directory(&docs_dir, &re)
-        .expect("Failed to extract snforge command snippets");
+    let snippet_type = SnippetType::forge();
 
-    // TODO(#2684)
-    let skipped_args = [
-        // for some reason `try_parse_from` fails on `--version` flag
-        vec!["snforge", "--version"],
-    ];
+    let snippets = extract_snippets_from_directory(&docs_dir, &snippet_type)
+        .expect("Failed to extract command snippets");
 
     for snippet in &snippets {
         let args = snippet.to_command_args();
-        let args: Vec<&str> = args.iter().map(String::as_str).collect();
+        let mut args: Vec<&str> = args.iter().map(String::as_str).collect();
 
-        if skipped_args.contains(&args) {
-            print_skipped_snippet_message(snippet, "snforge");
+        if snippet.config.ignored.unwrap_or(false) {
+            print_skipped_snippet_message(snippet);
             continue;
         }
 
-        let parse_result = Cli::try_parse_from(args);
+        let parse_result = Cli::try_parse_from(args.clone());
         let err_message = if let Err(err) = &parse_result {
             err.to_string()
         } else {
             String::new()
         };
-        assert_valid_snippet(parse_result.is_ok(), snippet, "snforge", &err_message);
+
+        assert_valid_snippet(parse_result.is_ok(), snippet, &err_message);
+
+        // Remove "snforge" from the args
+        args.remove(0);
+
+        // Remove "test" from the args
+        args.retain(|element| element != &"test");
```

Comment:
What happens if the snippet is `snforge test ...`? Does this just fail or do we handle it somewhere else.

Reply:
I updated it to use `runner` instead of `test_runner`.